core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -518,29 +529,30 @@ private[log] class Cleaner(val id: Int,
 
   private def compressMessages(buffer: ByteBuffer,
                                compressionCodec: CompressionCodec,
-                               messageFormatVersion: Byte,
                                messageAndOffsets: Seq[MessageAndOffset]) {
     require(compressionCodec != NoCompressionCodec, s"compressionCodec must not be $NoCompressionCodec")
     if (messageAndOffsets.nonEmpty) {
       val messages = messageAndOffsets.map(_.message)
       val magicAndTimestamp = MessageSet.magicAndLargestTimestamp(messages)
+      val magic = magicAndTimestamp.magic
```

Comment:
It may be worth adding a comment here that by using the magic from an inner message we implicitly fix the inconsistency introduced by KAFKA-4298. Just to make sure that someone reading this method also understands what's happening.